vsts_cd_manager/continuous_delivery_manager.py

```diff
@@ -155,20 +155,37 @@ def _verify_vsts_parameters(self, cd_account, source_repository):
         if source_repository.type in ['Github', 'ExternalGit'] and not cd_account:
             raise RuntimeError('You must provide a value for cd-account since your repo-url is not a Team Services repository.')
 
-    def _get_build_configuration(self, app_type, working_directory):
+    def _get_build_configuration(self, app_type_details, working_directory):
+
+        accepted_app_type = ['AspNetWap', 'AspNetCore', 'NodeJS', 'PHP', 'Python']
+        accepted_nodejs_task_runner = ['None', 'Gulp', 'Grunt']
+        accepted_python_framework = ['Bottle', 'Django', 'Flask']
+        accepted_python_version = ['Python 2.7.12 x64', 'Python 2.7.12 x86', 'Python 2.7.13 x64', 'Python 2.7.13 x86', 'Python 3.5.3 x64', 'Python 3.5.3 x86', 'Python 3.6.0 x64', 'Python 3.6.0 x86', 'Python 3.6.2 x64', 'Python 3.6.1 x86']
+        
         build_configuration = None
-        if app_type == 'AspNetWap':
-            build_configuration = BuildConfiguration(app_type, working_directory)
-        elif app_type == 'AspNetCore':
-            build_configuration = BuildConfiguration(app_type, working_directory)
-        elif app_type == 'PHP':
+        app_type = app_type_details.cd_app_type
+        if (app_type == 'AspNetWap') or (app_type == 'AspNetCore') or (app_type == 'PHP') :
             build_configuration = BuildConfiguration(app_type, working_directory)
-        elif app_type == 'NodeJSWithGulp':
-            build_configuration = BuildConfiguration('NodeJS', working_directory, 'Gulp')
-        elif app_type == 'NodeJSWithGrunt':
-            build_configuration = BuildConfiguration('NodeJS', working_directory, 'Grunt')
+        elif app_type == 'NodeJS' :
+            nodejs_task_runner = app_type_details.nodejs_task_runner
+            if any(s == nodejs_task_runner for s in accepted_nodejs_task_runner) :
+                build_configuration = BuildConfiguration(app_type, working_directory, nodejs_task_runner)
+            else:
+                raise RuntimeError("The nodejs_task_runner %s was not understood. Accepted values: %s." % (nodejs_task_runner, accepted_nodejs_task_runner))
```

Comment:
I don't know the best way to localize strings in Python. You may want to look at how it is done in the Azure CLI and try to model that. My guess is that it may not be fully localized since it uses other 3rd party packages.